packages/NodeCollector/NodeCollector/NodeRepository.php

```diff
@@ -499,6 +544,50 @@ public function findClassLike(string $classLikeName): ?ClassLike
         );
     }
 
+    /**
+     * @param Stmt[] $nodes
+     */
+    private function getClassMethodFromNodes(
+        array $nodes,
+        ClassReflection $classReflection,
+        string $className,
+        string $methodName
+    ): ?ClassMethod {
+        $reflectionClass = $classReflection->getNativeReflection();
+        $shortName = $reflectionClass->getShortName();
+
+        foreach ($nodes as $node) {
+            if ($node instanceof Namespace_) {
+                /** @var Stmt[] $nodeStmts */
+                $nodeStmts = $node->stmts;
+                $classMethod = $this->getClassMethodFromNodes($nodeStmts, $classReflection, $className, $methodName);
+
+                if ($classMethod instanceof ClassMethod) {
+                    return $classMethod;
+                }
+            }
+
+            $classMethod = $this->getClassMethod($node, $shortName, $methodName);
+            if ($classMethod instanceof ClassMethod) {
+                return $classMethod;
+            }
+        }
+
+        return null;
+    }
```

Comment:
NodeRepository should only work with nodes in analysed project.

For parsing vendor/Reflection code, the `\Rector\Core\Reflection\FunctionLikeReflectionParser` should be used instead. Could you move the logic there and re-use content in that service?

Reply:
I will try.

Reply:
implemented, moved to FunctionLikeReflectionParser https://github.com/rectorphp/rector/pull/6256/commits/677613048870fcdca38bd98fb5701cb791fea3ab